services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/events/LaneDirectionOfTravelEvent/LaneDirectionOfTravelEventRepositoryImpl.java

```diff
@@ -0,0 +1,167 @@
+
+    private final Quantity<Length> one_centimeter = Quantities.getQuantity(1, MetricPrefix.CENTI(Units.METRE));
+    private final Double ONE_CENTIMETER_IN_FEET = one_centimeter.to(USCustomary.FOOT).getValue().doubleValue();
+
+    private String collectionName = "CmLaneDirectionOfTravelEvent";
+
+    public Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest) {
+        Query query = new Query();
+
+        if (intersectionID != null) {
+            query.addCriteria(Criteria.where("intersectionID").is(intersectionID));
+        }
+        Date startTimeDate = new Date(0);
+        Date endTimeDate = new Date();
+
+        if (startTime != null) {
+            startTimeDate = new Date(startTime);
+        }
+        if (endTime != null) {
+            endTimeDate = new Date(endTime);
+        }
+
+        query.addCriteria(Criteria.where("eventGeneratedAt").gte(startTimeDate).lte(endTimeDate));
+        if (latest) {
+            query.with(Sort.by(Sort.Direction.DESC, "eventGeneratedAt"));
+            query.limit(1);
+        }else{
+            query.limit(props.getMaximumResponseSize());
+        }
+
+
+        return query;
+    }
+
+    public long getQueryResultCount(Query query) {
+        return mongoTemplate.count(query, LaneDirectionOfTravelEvent.class, collectionName);
+    }
+
+    public long getQueryFullCount(Query query){
+        int limit = query.getLimit();
+        query.limit(-1);
+        long count = mongoTemplate.count(query, LaneDirectionOfTravelEvent.class, collectionName);
+        query.limit(limit);
+        return count;
+    }
+
+    public List<LaneDirectionOfTravelEvent> find(Query query) {
+        return mongoTemplate.find(query, LaneDirectionOfTravelEvent.class, collectionName);
+    }
+
+    public List<IDCount> getLaneDirectionOfTravelEventsByDay(int intersectionID, Long startTime, Long endTime){
+        Date startTimeDate = new Date(0);
+        Date endTimeDate = new Date();
+
+        if (startTime != null) {
+            startTimeDate = new Date(startTime);
+        }
+        if (endTime != null) {
+            endTimeDate = new Date(endTime);
+        }
+
+        Aggregation aggregation = Aggregation.newAggregation(
+            Aggregation.match(Criteria.where("intersectionID").is(intersectionID)),
+            Aggregation.match(Criteria.where("eventGeneratedAt").gte(startTimeDate).lte(endTimeDate)),
+            Aggregation.project()
+                .and(DateOperators.DateToString.dateOf("eventGeneratedAt").toString("%Y-%m-%d")).as("dateStr"),
+            Aggregation.group("dateStr").count().as("count")
+        );
+
+        AggregationResults<IDCount> result = mongoTemplate.aggregate(aggregation, collectionName, IDCount.class);
+        List<IDCount> results = result.getMappedResults();
+
+        return results;
+    }
+
+    public List<IDCount> getMedianDistanceByFoot(int intersectionID, long startTime, long endTime){
```

Comment:
question: I'm a bit confused on what this method is doing, specifically in regards to the aggregation around medianDistanceFromCenterline and multiplying that by the ONE_CENTIMETER_IN_FEET. Is that medianXXX in centimeters and we want to return a foot value? Can we get some documentation in here to help explain?

Reply:
Added a javadoc comment to the parent method